### PR TITLE
Add prettier and eslint

### PR DESCRIPTION
**Pull request by sebastianbenz**
* Ignores archived samples
* Uses eslint/recommended rules
* Runs prettier and eslint (including --fix) pre-commit via husky
* Adds new npm scripts: 'lint', 'lint:fix' and 'prettier'
* Does not lint inline js code

For easier review I've split this PR into two commits, one for the configuration files and that fixes all prettier/eslint issues.

### CODE REVIEW

**Review by patrickkettner — APPROVED** [review]
lgtm

File: package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "devDependencies": {
+    "eslint": "^8.34.0",
+    "eslint-config-prettier": "8.6.0",
+    "eslint-plugin-prettier": "4.2.1",
+    "husky": "^8.0.0",
+    "lint-staged": "^13.1.2",
+    "prettier": "2.8.4"
+  },
+  "lint-staged":{
+    "**/*.{js}":[
```
Comment (oliverdunk) [inline package.json:32]: I attempted a commit and got:

```
⚠ Detected incorrect braces with only single value: `**/*.{js}`. Reformatted as: `**/*.js`
```

Changing this line to `"**/*.js":[` helped. Do you seem the same? 🤔

**Review by oliverdunk — APPROVED** [review]
Left a few nits but beyond that this is looking great! Thanks so much for working on it, I'm super excited to finally have some formatting 😁

File: README.md [hunk 2]
```diff
@@ -119,3 +119,18 @@ Read more on [Development Basics](https://developer.chrome.com/docs/extensions/m
     </tr>
   </tbody>
 </table>
+
+## Setting up your Environment
+
+If you want to contribute to this repository, you need to first [create your own fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+After forking chrome-extensions-samples to your own Github account, run the following steps to get started:
+
+```
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply README.md]: done

File: package.json (hunk 1 above)
Comment (sebastianbenz) [reply package.json]: fixed

Comment (sebastianbenz) [reply package.json]: Fixed

File: api-samples/alarms/index.js [hunk 3]
```diff
@@ -13,26 +13,26 @@ const pad = (val, len = 2) => val.toString().padStart(len, '0');
 
 // DOM event bindings
 
-//// Alarm display buttons
+// // Alarm display buttons
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply api-samples/alarms/index.js]: Fixed

File: README.md [hunk 4]
```diff
@@ -119,3 +119,18 @@ Read more on [Development Basics](https://developer.chrome.com/docs/extensions/m
     </tr>
   </tbody>
 </table>
+
+## Setting up your Environment
+
+If you want to contribute to this repository, you need to first [create your own fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply README.md]: Good idea, done - also updated the contributing text to use our standard template.

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+_archive
+third-party
+node_modules
+
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply .prettierignore]: fixed

**Review by sebastianbenz — COMMENTED** [review]
Thanks for the review - addressed all the comments.

**Comment by kirstyjackson244-png** [comment]
Wtf is this

**Pull request merged**